core/src/main/scala/org/apache/spark/deploy/master/Master.scala

```diff
@@ -539,7 +539,7 @@ private[deploy] class Master(
 
   private def completeRecovery() {
     // Ensure "only-once" recovery semantics using a short synchronization period.
-    if (state != RecoveryState.RECOVERING) { return }
+    if (state != RecoveryState.RECOVERING && state != RecoveryState.COMPLETING_RECOVERY) { return }
```

Comment:
Thank you for your review and suggest. The last change cannot work as I tested today.  I thought completeRecovery would be called again when some workers or drivers response to MasterChanged,  then the master(state is RecoveryState.COMPLETING_RECOVERY) will have the chance to complete the completeRecovery method and change state to ALIVE . I tested , but find no called again after the exception(maybe workers or drivers already response to MasterChanged before the exception).